File: src/main/java/org/minturtle/careersupport/codereview/entity/CommitPinpoint.java
package org.minturtle.careersupport.codereview.entity;

import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Document(collection = "commit_pinpoint")
public class CommitPinpoint {
    @Id
    private String id;

    private String lastSha;

    private Integer prNumber;

    private String repositoryName;

    @Builder.Default
    private Instant createdAt = Instant.now();

    public void upDateSha(String newSha) {
        this.lastSha = newSha;
    }
}

File: src/main/kotlin/org/minturtle/careersupport/codereview/entity/CommitPinpoint.kt
package org.minturtle.careersupport.codereview.entity

import org.minturtle.careersupport.common.utils.NanoIdGenerator
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.Instant


@Document(collection = "commit_pinpoint")
class CommitPinpoint(
    @Id
    val id: String = NanoIdGenerator.createNanoId(idSize = 10),
    var lastSha: String,
    val prNumber: Int,
    val repositoryName: String,
    val createdAt: Instant = Instant.now()
)